src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -405,6 +403,178 @@ public void insertInvalidVectorData() {
           .body("errors[0].errorCode", is("SHRED_BAD_VECTOR_VALUE"))
           .body("errors[0].exceptionClass", is("JsonApiException"));
     }
+
+    @Test
+    public void insertSimpleBinaryVector() {
+      final String id = UUID.randomUUID().toString();
+      final float[] expectedVector = new float[] {0.25f, 0.25f, 0.25f, 0.25f, 0.25f};
+      final String binaryVector = generateBase64EncodedBinaryVector(expectedVector);
+      String doc =
+              """
+                  {
+                    "_id": "%s",
+                    "name": "aaron",
+                    "$vector": {"$binary": "%s"}
+                  }
+              """
+              .formatted(id, binaryVector);
+
+      // insert the document
+      given()
+          .headers(getHeaders())
+          .contentType(ContentType.JSON)
+          .body("{ \"insertOne\": { \"document\": %s }}".formatted(doc))
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.insertedIds[0]", is(notNullValue()))
+          .body("data", is(nullValue()))
+          .body("errors", is(nullValue()));
+
+      // get the document and verify the vector value
+      Response response =
+          given()
+              .headers(getHeaders())
+              .contentType(ContentType.JSON)
+              .body("{\"find\": { \"projection\" : {\"$vector\" : 1}}}")
```

Comment:
Shouldn't this use "filter" to find actual document inserted? I guess it maybe happens to get right one as first but that's not guaranteed?